src/logic/safe/store/actions/createTransaction.ts

```diff
@@ -78,181 +75,253 @@ const navigateToTx = (safeAddress: string, txDetails: TransactionDetails) => {
   history.push(txRoute)
 }
 
-export const createTransaction =
-  (
-    {
-      safeAddress,
-      to,
-      valueInWei,
-      txData = EMPTY_DATA,
-      notifiedTransaction,
-      txNonce,
-      operation = Operation.CALL,
-      navigateToTransactionsTab = true,
-      origin = null,
-      safeTxGas: safeTxGasArg,
-      ethParameters,
-      delayExecution = false,
-    }: CreateTransactionArgs,
-    onUserConfirm?: ConfirmEventHandler,
-    onError?: ErrorEventHandler,
-  ): CreateTransactionAction =>
-  async (dispatch: Dispatch, getState: () => AppReduxState): Promise<DispatchReturn> => {
-    const state = getState()
+const getSafeTxGas = async (txProps: RequiredTxProps, safeVersion: string): Promise<string> => {
+  const estimationProps: SafeTxGasEstimationProps = {
+    safeAddress: txProps.safeAddress,
+    txData: txProps.txData,
+    txRecipient: txProps.to,
+    txAmount: txProps.valueInWei,
+    operation: txProps.operation,
+  }
 
-    const ready = await onboardUser()
-    if (!ready) return
+  let safeTxGas = '0'
+  try {
+    safeTxGas = await estimateSafeTxGas(estimationProps, safeVersion)
+  } catch (err) {
+    logError(Errors._617, err.message)
+  }
+  return safeTxGas
+}
 
-    const { account: from, hardwareWallet, smartContractWallet } = providerSelector(state)
-    const safeVersion = currentSafeCurrentVersion(state) as string
-    const safeInstance = getGnosisSafeInstanceAt(safeAddress, safeVersion)
-    const chainId = currentChainId(state)
+export class TxSender {
```

Comment:
Maybe we could move this to it's own file seeing as it's used for creation and processing?